### PR TITLE
introduce a prio_index_t type

### PR DESCRIPTION
**Pull request by arvidn**
for indexing into the priority sorted piece list, m_pieces, in piece_picker

### CODE REVIEW

File: include/libtorrent/piece_picker.hpp [hunk 1]
```diff
@@ -569,19 +572,29 @@ namespace libtorrent
...
+			prio_index_t index;
 
 #ifdef TORRENT_DEBUG_REFCOUNTS
 			// all the peers that have this piece
 			std::set<const torrent_peer*> have_peers;
 #endif
 
+#ifdef _MSC_VER
+#pragma warning(push, 1)
+#pragma warning(disable : 4268)
```
Comment (arvidn) [inline include/libtorrent/piece_picker.hpp]: as far as I can tell, all of these warnings are false positives and should probably be disabled at the project level

**Comment by arvidn** [comment]
@aldenml not sure if you'd like a chance to take a look. this is more of the same, but just contained to the piece picker internals

File: include/libtorrent/piece_picker.hpp (hunk 1 above)
Comment (aldenml) [inline include/libtorrent/piece_picker.hpp]: Probably, this is more of an informational(not problematic ) warning, right?

File: src/piece_picker.cpp [hunk 2]
```diff
@@ -1429,58 +1424,72 @@ namespace libtorrent
...
 
-		int start = 0;
-		for (int b : m_priority_boundaries)
+		prio_index_t start(0);
+		for (auto b : m_priority_boundaries)
 		{
 			if (start == b) continue;
-			aux::random_shuffle(&m_pieces[0] + start, &m_pieces[0] + b);
+			auto r = range(m_pieces, start, b);
+			aux::random_shuffle(r.begin(), r.end());
```
Comment (aldenml) [inline src/piece_picker.cpp]: Nice

**Comment by aldenml** [comment]
It looks very good to me.

sorry for the delay, I'm only with my phone for a few days :)

**Pull request merged**